### ISSUE TITLE
findByIdAndDelete should not be typed as returning ModifyResult

### ISSUE THREAD

**Issue by lmarinborges** [Automattic/mongoose#14233]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

8.0.3

### Node.js version

20.9.0

### MongoDB server version

6.0.12

### Typescript version (if applicable)

5.3.3

### Description

`findByIdAndDelete` is still defined in TypeScript as returning `ModifyResult`; unlike `findOneAndDelete`, which is properly typed as returning the deleted document (#14153).

### Steps to Reproduce

```ts
const Foo = mongoose.model(new Schema({
  field: String
}));

const foo: Foo | null = await Foo.findByIdAndDelete("1234"); // <- TypeScript error
```

### Expected Behavior

`findByIdAndDelete` should not return `ModifyResult`.

**Comment by lmarinborges** [comment]
Great, waiting for the next patch then, thank you 🙏🏼

**Comment by cbnsndwch** [comment]
I see a new `v7.6.8` package version in [NPM](https://www.npmjs.com/package/mongoose/v/7.6.8) but it didn't have the fix. Will this get backported? We're stuck on v7.x for the foreseeable future until we can migrate from NestJS 9 to 10

**Comment by lmarinborges** [comment]
@cbnsndwch

**Comment by lmarinborges** [comment]
@cbnsndwch You could try to temporarily solve this issue by using [`pnpm patch`](https://pnpm.io/cli/patch) or the [`patch-package`](https://www.npmjs.com/package/patch-package) utility to apply the changes made in #14153; however I am not aware of how different v7 is to V8, it might break some stuff.

Otherwise you can just use findOneAndDelete instead, which seems to be properly typed.

**Comment by cbnsndwch** [comment]
@lmarinborges thanks for the suggestion! 

The types in Mongoose are a not trivial to detangle and I have just three places in my app where this creates an issue so I just `any`-ed the Promise for now and added an note internally to document the why. It felt less disruptive this way.

The long term solution for us is likely migrating to Nest 10 and Mongoose v8 anyway, so this will serve as yet another incentive 😅

**Comment by cbnsndwch** [comment]
> Otherwise you can just use findOneAndDelete instead, which seems to be properly typed.

Not seeing this on my end. Note, I'm on `v7.6.8`.

![image](https://github.com/Automattic/mongoose/assets/8313760/5e3c21f1-f0a1-49ea-bc16-cb849b8e8c89)

**Comment by vkarpov15** [comment]
@cbnsndwch this fix is only in 8.0.4, not on 7.6.8 currently. We will backport.

**Comment by cbnsndwch** [comment]
@vkarpov15 thank you for the reply!

I've started migrating my projects to Nest 10.x and Mongoose 8.x now, so I'm halfway GTG. Glad to hear you're backporting the fix though, thanks 🙏!

**Comment by vkarpov15** [comment]
I cherry-picked 775aadf16be64ffc19f86aa05797095ad3d9eca3 and 7f9406ff44c5d8536599060c3ae38009f866ab60 into 7.x, so this fix will be released with 7.6.9 :+1:
